database/redis/metric.go

```diff
@@ -106,6 +106,7 @@ func (connector *DbConnector) SaveMetrics(metrics map[string]*moira.MatchedMetri
 	ctx := connector.context
 
 	rand.Seed(time.Now().UnixNano())
+	pipe := c.TxPipeline()
```

Comment:
Would you please explain, why you use pipe here? Is it perfomance optimization?

Reply:
Yes it's a performance optimization. Pipe collects records and send them together to redis. That model minimizes internet communicate overheads.

Reply:
I don't like pipe because people use it everywhere even it is not need like premature optimization. And in Moira pipe is used everywhere, it is mixed up with business logic pretty often. So it makes fragile architecture, mixed data access layer and business logic layer. But it is IMHO, nevermind, just my thoughs